File: Daria-Goldenberg/SecondTerm/Task 7/WeatherGetter/Weather.cs
namespace WeatherGetter
{
	public class Weather
	{
		public readonly double? TemperatureCelsius;
		public readonly double? TemperatureFahrenheit;
		public readonly double? CloudCover;
		public readonly double? Humidity;
		public readonly double? WindSpeed;
		public readonly string? WindDirection;
		public readonly string? Precipitation;

		public Weather(double? temperatureCelsius, double? temperatureFahrenheit, double? cloudCover,
						double? humidity, double? windSpeed, string? windDirection, string? precipitation)
		{
			TemperatureCelsius = temperatureCelsius;
			TemperatureFahrenheit = temperatureFahrenheit;
			CloudCover = cloudCover;
			Humidity = humidity;
			WindSpeed = windSpeed;
			WindDirection = windDirection;
			Precipitation = precipitation;
		}
	}
}

File: IvanShurenkov/Second Term/Block3/Task2/lib/weather/src/main/kotlin/lib/weather/date/Weather.kt
package lib.weather.date

class Weather {
    var temperature: Temperature? = null
    var cloudCoverage: CloudCoverage? = null
    var humidity: Humidity? = null
    var precipitation: Precipitation? = null
    var windSpeed: WindSpeed? = null
    var windDirection: WindDirection? = null

    override fun toString(): String {
        var ret = "Temperature: " + if (temperature != null) {
            "${temperature!!.celsius}°C ${temperature!!.fahrenheit}°F"
        } else "No date available"
        ret += "\nCloud coverage: " + if (cloudCoverage != null) "${cloudCoverage!!.percent}"
        else "No date available"
        ret += "\nHumidity: " + if (humidity != null) "${humidity!!.percent}"
        else "No date available"
        ret += "\nPrecipitation: " + if (precipitation != null) "${precipitation!!.mmPerHour}"
        else "No date available"
        ret += "\nWind speed: " + if (windSpeed != null) "${windSpeed!!.speed}"
        else "No date available"
        ret += "\nWind direction: " + if (windDirection != null) "${windDirection!!.degree}°"
        else "No date available"
        return ret
    }
}